crates/chain/src/keychain/txout_index.rs

```diff
@@ -133,8 +165,12 @@ impl<K> KeychainTxOutIndex<K> {
     }
 }
 
+/// Methods that are *re-exposed* from the internal [`SpkTxOutIndex`].
 impl<K: Clone + Ord + Debug> KeychainTxOutIndex<K> {
     /// Return a reference to the internal [`SpkTxOutIndex`].
+    ///
+    /// **WARNING:** The internal index will contain lookahead spks. Refer to
+    /// [struct-level docs](KeychainTxOutIndex) for more about `lookahead`.
```

Comment:
I would assume in some cases getting lookahead spks will be handy.

I.e. we haven't finalized how a CBF chain-source will look like and how it will interact with `bdk_chain` structures. However, in my head, it makes sense that you will want to try match lookahead spks against GCS (filters).

Maybe later on, we determine that it will make sense to use helper methods on `KeychainTxOutIndex` instead. However, for most usecases now, outputting lookahead spks is almost always non-intentional.

Reply:
In summary, I think we should keep `.inner` with a warning.